server/middleware/fetchOffenderList.test.ts

```diff
@@ -0,0 +1,89 @@
+import { getMockReq, getMockRes } from '@jest-mock/express'
+import { when } from 'jest-when'
+import PrisonerSearchApiClient from '../data/prisonerSearchApiClient'
+import PrisonApiClient from '../data/prisonApiClient'
+import PrisonRegisterApiClient from '../data/prisonRegisterApiClient'
+import WhereaboutsApiClient from '../data/whereaboutsApiClient'
+import PrisonService from '../services/prisonService'
+import atLeast from '../../jest.setup'
+import inmateDetails1 from './fixtures/inmate_details_1.json'
+import fetchOffenderList from './fetchOffenderList'
+
+jest.mock('../services/prisonService')
+jest.mock('../data/prisonApiClient')
+jest.mock('../data/prisonerSearchApiClient')
+jest.mock('../data/prisonRegisterApiClient')
+jest.mock('../data/whereaboutsApiClient')
+
+describe('fetchOffenderList', () => {
+  const prisonApiClient = new PrisonApiClient() as jest.Mocked<PrisonApiClient>
+  const prisonerSearchApiClient = new PrisonerSearchApiClient() as jest.Mocked<PrisonerSearchApiClient>
+  const prisonRegisterApiClient = new PrisonRegisterApiClient() as jest.Mocked<PrisonRegisterApiClient>
+  const whereaboutsApiClient = new WhereaboutsApiClient() as jest.Mocked<WhereaboutsApiClient>
+
+  const prisonService = new PrisonService(
+    prisonApiClient,
+    prisonerSearchApiClient,
+    prisonRegisterApiClient,
+    whereaboutsApiClient,
+  )
+
+  it('Success', async () => {
+    const req = getMockReq({
+      query: {
+        prisonId: 'MDI',
+      },
+      session: {
+        user: {
+          token: 'token',
+          activeCaseLoad: { caseLoadId: 'MDI' },
+        },
+      },
+    })
+    const { res } = getMockRes({
+      locals: {
+        user: { token: 'token', activeCaseLoad: { caseLoadId: 'MDI' } },
+      },
+    })
+    const next = jest.fn()
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
Its the old problem with the api types not matching what we actually get back from the api. In the past i've created another version of the type, but this issue only manifests in tests so on this occasion i thought id ignore rather than duplicate the type with optional params

Reply:
yeah its annoying - happy to take your steer on best solution - ghost types or ts-ignore